src/EditorFeatures/CSharp/FixInterpolatedVerbatimString/FixInterpolatedVerbatimStringCommandHandler.cs

```diff
@@ -0,0 +1,66 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System;
+using System.ComponentModel.Composition;
+using Microsoft.CodeAnalysis.CSharp;
+using Microsoft.CodeAnalysis.Editor.Shared.Extensions;
+using Microsoft.CodeAnalysis.Text;
+using Microsoft.VisualStudio.Commanding;
+using Microsoft.VisualStudio.Text.Editor.Commanding.Commands;
+using Microsoft.VisualStudio.Utilities;
+using VSCommanding = Microsoft.VisualStudio.Commanding;
+
+namespace Microsoft.CodeAnalysis.Editor.CSharp.FixInterpolatedVerbatimString
+{
+    [Export(typeof(VSCommanding.ICommandHandler))]
+    [ContentType(ContentTypeNames.CSharpContentType)]
+    [Name(nameof(FixInterpolatedVerbatimStringCommandHandler))]
+    internal sealed class FixInterpolatedVerbatimStringCommandHandler : IChainedCommandHandler<TypeCharCommandArgs>
+    {
+        public string DisplayName => CSharpEditorResources.Fix_interpolated_verbatim_string;
+
+        public void ExecuteCommand(TypeCharCommandArgs args, Action nextCommandHandler, CommandExecutionContext executionContext)
+        {
+            // We need to check for the token *after* the opening quote is typed, so defer to the editor first
+            nextCommandHandler();
+
+            if (args.TypedChar == '"')
+            {
+                var caret = args.TextView.GetCaretPoint(args.SubjectBuffer);
+                if (caret != null)
+                {
+                    var position = caret.Value.Position;
+                    var snapshot = caret.Value.Snapshot;
+
+                    if (position >= 3 &&
+                        snapshot[position - 1] == '"' &&
+                        snapshot[position - 2] == '$' &&
+                        snapshot[position - 3] == '@')
+                    {
+                        var document = snapshot.GetOpenDocumentInCurrentContextWithChanges();
+                        if (document != null)
+                        {
+                            var root = document.GetSyntaxRootSynchronously(executionContext.OperationContext.UserCancellationToken);
+                            var token = root.FindToken(position - 3);
+                            if (token.IsKind(SyntaxKind.InterpolatedVerbatimStringStartToken))
+                            {
+                                var newToken = SyntaxFactory.Token(
+                                    token.LeadingTrivia,
+                                    SyntaxKind.InterpolatedVerbatimStringStartToken,
+                                    "$@\"",
+                                    "$@\"",
+                                    token.TrailingTrivia);
+
+                                var newDocument = document.WithSyntaxRoot(root.ReplaceToken(token, newToken));
+                                document.Project.Solution.Workspace.TryApplyChanges(newDocument.Project.Solution);
```

Comment:
i would not do this.  I woudl simply edit the buffer directly.

Reply:
➡️ Fixed in 946f89ea573d3563c0202aaba86bc616ca0e36c1